src/Compilers/CSharp/Portable/BoundTree/ConversionGroup.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// A group is a common instance referenced by all BoundConversion instances
+    /// generated from a single Conversion. The group is used by NullableWalker to
+    /// determine which BoundConversion nodes should be considered as a unit.
+    /// </summary>
+    [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
+    internal sealed class ConversionGroup
```

Comment:
>ConversionGroup [](start = 26, length = 15)

I am suspicious of this approach. I don't believe that the grouping of conversions, for example around a user-defined conversion, have any semantic meaning. If this is used to affect the interpretation of the meaning of the bound nodes, then I suspect there is a bug somewhere. I would suggest (now or after this PR is integrated) investigating trying to get rid of this to see if it is really needed to make any tests pass.

Reply:
Logged [https://github.com/dotnet/roslyn/issues/27163](https://github.com/dotnet/roslyn/issues/27163).

---
In reply to: [190725973](https://github.com/dotnet/roslyn/pull/26356#discussion_r190725973) [](ancestors = 190725973)